main.cpp

```diff
@@ -1,478 +1,571 @@
+        std::string fingerboardWoodType = "maple";
+        std::string fretIndicatorInlayType = "shell dot";
+
+        void supplyFingerRest(int neckBoltPosition);
+        void helpIndexingNotes(SongWriter classicalWriterA, bool fretInlay = true);
+        void holdStringsInPosition(std::string nutType = "cow bone");
+    };
+
+    struct Body
+    {
+        std::string bodyShape = "strat";
+        int numberOfPickups = 3;
+        int numberOfPots = 3;
+        std::string woodType= "alder";
+        int thicknessInMM = 40;
+    
+        void holdPartsTogether(std::string boltType = "bolt on");
+        void improveTheVibration(float resonateFrequency = 50);
+        void makeGuitarLookCool (SongWriter rockWriterA, bool colourfulFinish = true);
+    };
+
+    struct Bridge
+    {
+        int numberOfStringSlot = 6;
+        std::string materialType = "stainless steel";
+        std::string color = "nickel";
+        std::string whammyBarType = "floyd rose";
+        int numberOfBodyScrewHole = 2;
+
+         void holdStringsStill(bool stringInTune = true);
+        void makePitchDive(float stringTensionInKg = 1);
+
+        //returns if the bridge matches guitar body's color
+        bool matchTheBodyColor(SongWriter rockWriterA);
+    };
+
+    struct String
+    {
+        std::string woundType = "round wound";
+        float firstStringGaugeInInch = 0.090f;
+        float sixthStringGaugeInInch = 0.042f;
+        std::string brand = "Elixir";
+        std::string coatType = "ultra-thin";
+
+        struct StringType
+        {
+            bool firstStringWound = false;
+            bool secondStringWound = false;
+            bool thirdStringWound = true;
+            bool fourthStringWound = true;
+            bool fifthStringWound = true;
+            bool sixthStringWound = true;
+
+            //returns how many wound strings do we have
+            int checkWoundString(int woundStringAmount = 0);
+
+            //returns the proper string pack name
+            std::string chooseStringPack(int roundWoundStringNum = 3, int flatWoundStringNum = 4);
+
+            //returns how much would the proper string pack cost
+            float calculateStringPackBudget(float averageCostOfStringPack = 6);
+        };
+        
+        void vibrate(Body strat, float frequency = 440, std::string pitchNotation = "A4");
+        void getRust(bool colorChange = false);
+        void getBroken(int brokenStringAmount = 0);
+};
+
+    struct Pickup
+    {
+        std::string coilType = "single coil";
+        int stringDistanceInMm = 8;
+        std::string magnetMaterialType = "alnico 5";
+        std::string coverColor = "ivory";
+        int stringNumber = 6;
+
+        void pickupStringVibration(bool pluckString = false);
+        void changeTheGuitarTone(int pickupSwitchPosition = 5);
+
+        //returns if pickup cover color matches the body color
+        bool matchTheBodyColor(SongWriter rockWriterA);
+    };
+
```

Comment:
Delete all of this.
You already declared these classes earlier in the source file. 
you declare instances of THOSE classes.  
You don't copy those class definitions into this class.
- [ ] revise